### PR TITLE
add more mojito.js tests

### PR DESCRIPTION
**Pull request by isao**

### CODE REVIEW

File: lib/mojito.js [hunk 1]
```diff
@@ -149,6 +149,84 @@ MojitoServer.prototype._startupTime = null;
 //  Private Methods
 //  ---------------
 
+MojitoServer.prototype._makeMwList = function (app_mw, mojito_mw) {
```
Comment (caridy) [inline lib/mojito.js]: There is not need to shorten the names when it comes to nodejs. Let's use the full "Middleware" instead of "Mw".

Comment (caridy) [inline lib/mojito.js]: Let's also add the documentation for these two new methods.

File: lib/mojito.js [hunk 2]
```diff
@@ -282,13 +331,7 @@ MojitoServer.prototype._configureAppInstance = function(app, options) {
...
         });
         if (appConfig.debugMemory) {
-            outputHandler.setMemoryDebugObjects({
-                'YUI.Env': YUI.Env,
-                'YUI.config': YUI.config,
-                'YUI.process': YUI.process,
-                Y: Y,
-                store: store
-            });
+            outputHandler.setMemoryDebugObjects(debugConfig);
```
Comment (caridy) [inline lib/mojito.js]: We can eliminate the setMemoryDebugObject, we don't need that anymore.

File: lib/mojito.js [hunk 3]
```diff
@@ -151,6 +151,101 @@ MojitoServer.prototype._startupTime = null;
...
+            // backwards-compatibility: user-provided middleware is
+            // specified by path
+            midPath = libpath.join(midDir, midName);
+            //console.log("======== MIDDLEWARE user " + midPath);
+            midBase = libpath.basename(midPath);
+            if (0 === midBase.indexOf('mojito-')) {
+                // Same as above (case of Mojito's special middlewares)
+                // Gives a user-provided middleware access to the YUI
+                // instance, resource store, logger, context, etc.
+                //                                                ...dead code?
```
Comment (drewfish) [inline lib/mojito.js]: This isn't dead code, this is in fact the new API.  The "else" branch is actually the old code/API which we'll deprecate at some point.

File: tests/unit/lib/test-mojito.js [hunk 4]
```diff
@@ -250,5 +261,333 @@ YUI().use('mojito', 'test', function (Y) {
...
+
+        'test listen 1': function () {
+            var port = 1234,
+                host = 'letterman';
+
+            this._startupTime = +new Date();
+            this._options = {verbose: true};
+
+            Mojito.Server.prototype.listen.call(this, port, host);
+        },
```
Comment (drewfish) [inline tests/unit/lib/test-mojito.js:332]: Where are the asserts in this test?

File: tests/unit/lib/test-mojito.js [hunk 5]
```diff
@@ -250,5 +261,333 @@ YUI().use('mojito', 'test', function (Y) {
...
+            Y.Mock.expect(app, {
+                method: 'listen',
+                arguments: [port, host, cb]
+            });
+
+            this._startupTime = null;
+            this._options = {verbose: false};
+            this._app = app;
+
+            Mojito.Server.prototype.listen.call(this, port, host, cb);
```
Comment (drewfish) [inline tests/unit/lib/test-mojito.js]: This approach, of passing "this" to the function call, is weird.  Reading it, I don't understand what kind of object "this" is, or how it relates to the tests we're attempting.  If you'd created a mock object and passed it as the first argument to call() that'd make more sense to me.

Comment (isao) [inline tests/unit/lib/test-mojito.js]: agreed, that's much cleaner. I'll update, thx

**Comment by lzhan** [comment]
+1

**Pull request merged**